proof/dleq.go

```diff
@@ -0,0 +1,163 @@
+package proof
+
+import (
+	"errors"
+
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/hash"
+	"github.com/dedis/crypto/random"
+)
+
+// This package provides functionality to create and verify non-interactive
+// zero-knowledge (NIZK) proofs for the equality (EQ) of discrete logarithms (DL).
```

Comment:
doc: could you give some small explanation of the mathematics behind it? Just the formulas that are used to create and verify the proof.